week2/index.js

```diff
@@ -0,0 +1,135 @@
+
+const fs = require('fs');
+const readline = require('readline');
+let todos = [];
+
+
+if (process.argv[2] !== undefined) {
+    start();
+} else {
+    console.log('Please specify an option to use, here is the manual\n');
+    manualReader('./help.txt');
+}
+
+
+function start() {
+    switch (process.argv[2].trim().toLowerCase()) {
+        case 'help':
+            manualReader('./help.txt');
+            break;
+        case 'add':
+            checkFileContent().then((item) => addTodoItem(item)).catch((err) => console.log(err));
+            break;
+        case 'remove':
+            itemRemover(process.argv[3]);
+            break;
+        case 'list':
+            console.log('The items on your todo list: ');
+            todoFileReader('./todos.txt');
+            break;
+        case 'reset':
+            itemsResetter();
+            break;
+        default:
+            console.log('No such option! Take a look at the manual\n');
+            manualReader('./help.txt');
+            break;
+    }
+}
+
+
+function manualReader(fileName) {
+    fs.readFile(fileName, 'UTF-8', (err, data) => {
+        console.log(data);
+    });
+}
+
+//  display the todo file once called
+function todoFileReader(fileName) {
+    fs.exists(fileName, (exists) => {
+        //see if file exists to add the existing items to the array
+        if (exists) {
+            fs.readFile(fileName, 'UTF-8', (err, data) => {
+                if (data === "") {
+                    console.log('There are no items yet, try using \'node index.js add "some todo item"\'');
+                    return;
+                }
+                data = data.split('\r\n');
+                let text = data.reduce(((last, item, index) => last += `${index + 1}. ${item}\n`), ``);
+                console.log(text);
+            });
+        } else {
+            console.log('There are no items yet, try using \'node index.js add "some todo item"\'');
+        }
+    });
+}
+
+
+
+//check if there's already a file and to push the item in
+
+function checkFileContent() {
+    return new Promise(function (resolve, reject) {
+        fs.exists('./todos.txt', (exists) => {
+            if (!exists) {
+                resolve(process.argv[3]);
+                return;
+            }
+            fs.readFile('todos.txt', 'UTF-8', (err, data) => {
+                todos = data.trim().split('\r\n');
+                resolve(process.argv[3]);
+            });
+        });
+    });
+}
+
+//add the new item to the array
+function addTodoItem(todoItem) {
+    if (todoItem !== '') {
+        if (todoItem.trim() !== '') {
```

Comment:
Is this to check that the string isn't just whitespace?

First of all, you could reduce it to `if ( todoItem !== '' && todoItem.trim() !== '' )`. But since an empty string will still be an empty string after `trim()`, you could just leave out the first check and make it `if ( todoItem.trim() !== '' )`.

But you also need to trim the string again (below on line 90), so you're running the `trim()` function twice. Now, this isn't a big deal because trim is fast on a short string. But what if the string is 100M? Running `trim()` could become rather slow.
You could simplify this with doing the `trim` once (before the `if`) conditional, and then checking and reusing it, like:
```
let trimmedItem = todoItem.trim();
if ( trimmedItem !== '' ) {
  todos.push(trimmedItem);
  ...
}
```